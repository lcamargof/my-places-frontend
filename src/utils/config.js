export const MAPBOX_TOKEN = 'pk.eyJ1IjoibHVpc2t4IiwiYSI6ImNqZzc3OGhyOTFreG8zM2xvNDJ2MmE4amUifQ.-4-kEUuKIeCmvchZlLviKQ';
export const API_URL = process.env.NODE_ENV === 'development' ? 'http://localhost:5001' : '';
export const DEFAULT_COORDINATES = [-103.40932570000001, 20.6713195];