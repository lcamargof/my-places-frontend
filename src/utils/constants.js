export const ADD_ACTION = 'ADD';
export const EDIT_ACTION = 'EDIT';