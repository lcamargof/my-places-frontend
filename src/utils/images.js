const image = new Image();
image.src = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAQAAABLCVATAAABP0lEQVR4AeXUP0tCURjH8S8OhtKV/kwX13oPQatEEs3ZnhlFmXOt0VRzi70HQ0FoioiGcqjoDQi1pYHUcPX2NAU94D3nXB39fDd/j3fwgkwmjyJXtAkIaFNji2liS1Chg6D7oEyCGDJcIxE18XCU5h4xdEcKJ1XE0gUOlhFrPyxhVUdUfVo8ECCqGhbzDNQXGvgA+DT045nBaF2dP5PkT5IXta1hVFHHJbVtq+0Qo2N1nFNbTm1HGO2r46LaSmrbw2g18jea4lVtKxjNEQ59a1ma6vOQWSxuEVXAIy36iOoGq13EoR2sMvQQSz08HJwjls5w4vONGPrCx9EpYugEZx7vSERveMSwiURUIKYGMqQ6sWXpIug6ZCG+AoJugxFdqsdUGVn637/iEynGsMgngtBlgTHlCRmQh/GVOWCS/QJAbX2ObhdYKAAAAABJRU5ErkJggg==';
export const IMAGES = ['place', image];
