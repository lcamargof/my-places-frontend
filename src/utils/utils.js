export const getUserLocation = () => {
  return new Promise((resolve, reject) => {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(position => (
        resolve([position.coords.longitude, position.coords.latitude])
      ), () => reject());
    } else {
      reject();
    }
  });
};