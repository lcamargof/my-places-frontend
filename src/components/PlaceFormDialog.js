import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Dialog, DialogActions, DialogContent, DialogTitle, withStyles } from 'material-ui';
import { ADD_ACTION } from '../utils/constants';

const styles = () => ({

});

class PlaceFormDialog extends Component {

  render() {
    const { onClose, onSubmit, type, place, open } = this.props;

    const title = type === ADD_ACTION ? 'Add your place!' : 'Edit your place :)';

    return (
      <Dialog
        open={open}
        onClose={onClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
        <DialogContent>
          HOLI
        </DialogContent>
        <DialogActions>
          <Button onClick={onClose}>
            Cancel
          </Button>
          <Button onClick={onSubmit} color="primary">
            SUBMIT
          </Button>
        </DialogActions>
      </Dialog>
    );
  }
}

PlaceFormDialog.propTypes = {
  place: PropTypes.object,
  classes: PropTypes.object.isRequired,
  onClose: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

PlaceFormDialog.defaultProps = {
  place: {},
};

export default withStyles(styles)(PlaceFormDialog);