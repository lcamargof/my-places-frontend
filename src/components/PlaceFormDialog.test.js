import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ADD_ACTION, EDIT_ACTION } from '../utils/constants';
import { PlaceFormDialog } from './PlaceFormDialog';

const mockPlace = { name: 'test', open: '00:00', close: '23:59', description: 'test' };

describe('Component: PlaceFormDialog', () => {
  let component = null;
  var classes, place, onClose, onSubmit, type;

  beforeEach(() => {
    classes = {};
    place = undefined;
    onClose = jest.fn();
    onSubmit = jest.fn();
    type = EDIT_ACTION;
  });

  const buildComponent = () => {
    const props = {
      classes, place, onClose, onSubmit, type
    };

    return shallow(<PlaceFormDialog {...props} />);
  };

  describe('Snapshots', () => {
    it('Should render correctly (EDIT)', () => {
      component = buildComponent();
      expect(shallowToJson(component)).toMatchSnapshot();
    });

    it('Should render correctly (ADD)', () => {
      place = { location: []};
      type = ADD_ACTION;
      component = buildComponent();
      expect(shallowToJson(component)).toMatchSnapshot();
    });
  });

  describe('Actions', () => {
    it('Should trigger close when clicked the cancel button', () => {
      component = buildComponent();
      component.find('#close-place-form').simulate('click');
      expect(onClose).toHaveBeenCalledTimes(1);
    });

    it('Should trigger submit when clicked the delete button', () => {
      component = buildComponent();
      component.setProps({ place: mockPlace });
      component.find('#submit-place-form').simulate('click', { preventDefault: jest.fn() });
      expect(onSubmit).toHaveBeenCalledTimes(1);
    });

    it('Should trigger submit when the form is submitted', () => {
      component = buildComponent();
      component.setProps({ place: mockPlace });
      component.find('#place-form').simulate('submit', { preventDefault: jest.fn() });
      expect(onSubmit).toHaveBeenCalledTimes(1);
    });

    it('Should validate the form correctly', () => {
      component = buildComponent();
      component.setProps({ place: mockPlace });
      expect(component.instance().validation()).toBeTruthy();

      component.setProps({ place: Object.assign({}, mockPlace, { name: '' }) });
      expect(component.instance().validation()).toBeFalsy();
    });
  });
});