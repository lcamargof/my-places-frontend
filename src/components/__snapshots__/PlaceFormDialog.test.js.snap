// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PlaceFormDialog Snapshots Should render correctly (ADD) 1`] = `
<WithStyles(Dialog)
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  onClose={[Function]}
>
  <WithStyles(DialogTitle)
    id="alert-dialog-title"
  >
    Add your place!
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <form
      autoComplete="off"
      id="place-form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <TextField
        error={false}
        fullWidth={true}
        helperText=""
        id="name"
        label="Name"
        margin="normal"
        onChange={[Function]}
        placeholder="I'm sure it will be a cool name"
        required={false}
        select={false}
        value=""
      />
      <TextField
        error={false}
        fullWidth={true}
        helperText=""
        id="description"
        label="Description"
        margin="normal"
        multiline={true}
        onChange={[Function]}
        placeholder="Describe it!"
        required={false}
        select={false}
        value=""
      />
      <div>
        <TextField
          error={false}
          fullWidth={true}
          helperText=""
          id="open"
          label="Open"
          margin="normal"
          onChange={[Function]}
          placeholder="12:00"
          required={false}
          select={false}
          type="time"
          value="08:00"
        />
        <TextField
          error={false}
          fullWidth={true}
          helperText=""
          id="close"
          label="Close"
          margin="normal"
          onChange={[Function]}
          placeholder="23:00"
          required={false}
          select={false}
          type="time"
          value="20:00"
        />
      </div>
    </form>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="close-place-form"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      id="submit-place-form"
      onClick={[Function]}
    >
      SUBMIT
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`Component: PlaceFormDialog Snapshots Should render correctly (EDIT) 1`] = `
<WithStyles(Dialog)
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  onClose={[Function]}
>
  <WithStyles(DialogTitle)
    id="alert-dialog-title"
  >
    Edit your place :)
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <form
      autoComplete="off"
      id="place-form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <TextField
        error={false}
        fullWidth={true}
        helperText=""
        id="name"
        label="Name"
        margin="normal"
        onChange={[Function]}
        placeholder="I'm sure it will be a cool name"
        required={false}
        select={false}
        value=""
      />
      <TextField
        error={false}
        fullWidth={true}
        helperText=""
        id="description"
        label="Description"
        margin="normal"
        multiline={true}
        onChange={[Function]}
        placeholder="Describe it!"
        required={false}
        select={false}
        value=""
      />
      <div>
        <TextField
          error={false}
          fullWidth={true}
          helperText=""
          id="open"
          label="Open"
          margin="normal"
          onChange={[Function]}
          placeholder="12:00"
          required={false}
          select={false}
          type="time"
          value="08:00"
        />
        <TextField
          error={false}
          fullWidth={true}
          helperText=""
          id="close"
          label="Close"
          margin="normal"
          onChange={[Function]}
          placeholder="23:00"
          required={false}
          select={false}
          type="time"
          value="20:00"
        />
      </div>
    </form>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="close-place-form"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      id="submit-place-form"
      onClick={[Function]}
    >
      SUBMIT
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
